src/test/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannelTest.java

```diff
@@ -251,6 +251,71 @@ public void testCloseChannelException() throws Exception {
     topicPartitionChannel.closeChannel();
   }
 
+  @Test
+  public void closeChannel_withSFExceptionInFuture_swallowsException() {
+    // given
+    CompletableFuture<Void> closeChannelFuture = new CompletableFuture<>();
+    closeChannelFuture.completeExceptionally(SF_EXCEPTION);
+
+    Mockito.when(mockStreamingChannel.close()).thenReturn(closeChannelFuture);
+    Mockito.when(mockStreamingChannel.getFullyQualifiedName()).thenReturn(TEST_CHANNEL_NAME);
+
+    TopicPartitionChannel topicPartitionChannel =
+        new TopicPartitionChannel(
+            mockStreamingClient,
+            topicPartition,
+            TEST_CHANNEL_NAME,
+            TEST_TABLE_NAME,
+            true,
+            streamingBufferThreshold,
+            sfConnectorConfig,
+            mockKafkaRecordErrorReporter,
+            mockSinkTaskContext,
+            mockSnowflakeConnectionService,
+            new RecordService(),
+            mockTelemetryService,
+            false,
+            null);
+
+    // when
+    topicPartitionChannel.closeChannel();
```

Comment:
Adding `Assertions.assertDoesNotThrow` would make the reason of this test more explicit.